src/Request.php

```diff
@@ -176,16 +176,23 @@ public function handleError($error)
         ));
     }
 
+    public function handleClose()
+    {
+        $this->closeError(new \RuntimeException(
+            "Connection closed before receiving response"
+        ));
```

Comment:
:-1: This should probably stay in the `end` handler. The `end` event will be emitted when the connection has ended, the `close` event will also be emitted if you explicitly call `close()` in which case this would not be considered an error.